### PR TITLE
Change phone number edits

### LINKED ISSUE #689: Phone number 'Edit' link on User Profile not changing
Phone number 'Edit' link on User Profile is not changing to Save when you click on the phone number to edit it. It still works on the SLC instructors

### PR DESCRIPTION
**Pull request by AndersonStettner**
Issue: Make sure the behavior of phone number inputs are consistent across the application and make it so that if a user has made changes to a phone number and the click off without saving the phone number returns to how it was originally. 

________________________________________
Did:
- Created two new functions in base.js: setupPhoneNumber and processPhoneSetup.
- Changed the notification when the phone number is saved from a msgFlash to a msgToast.

________________________________________
Expected behavior when editing phone number:
-  On click of _"Edit"_ or on _focus_ of input: _"Edit"_ will become _"Save"_ and the phone input will be focused.
- On click of _"Save"_ with a VALID phone number: _"Save"_ will become _"Edit"_, the phone input will be un-focused, the new phone number will be saved, and a toast will appear in the bottom right confirming the new phone number has been saved.
- On click of _"Save"_ with an INVALID phone number: _"Save"_ will stay _"Save"_, the background of the phone input will turn red, and the phone input will stay focused.
- On _focus out_ of phone input without clicking _"Save"_: _"Save"_ will become _"Edit"_ and the contents of the phone input will restore the phone number to its original state before the input was focused. 

________________________________________
To test on User Profile:
- Make sure that the edit button behaves as expected when you are focusing/un-focusing the phone input.
- Delete part of the phone number then click out of the phone input without click the save button and make sure the phone number is restored to its origin state.
- Delete part of the phone number  and attempt to save it. It should not be allowed to save and the phone input bg color should turn red while still being focused. 
- Change the phone number and save it. Make sure the new phone number is actually saved and the msgToast pops up confirming the save. 

________________________________________
To test on SLC Instructor:
- Edit an SLC proposal and repeat all testing actions that were done on the User Profile on the Instructors phone number. 
- Add a second SLC instructor to the instructor table and make sure the phone numbers still behave as expected. 
- Remove all instructors from the instructor table and then add one back and make sure the phone number still behaves as expected.

### CODE REVIEW

**Comment by AndersonStettner** [comment]
[Phone number 'Edit' link on User Profile not changing #689](https://trello.com/c/JIFlAoYD/489-phone-number-edit-link-on-user-profile-not-changing-689)

**Comment by gahimbaref** [comment]
Everything works as expected. However, the toast messages blend in with the page color, someone might not notice them right away especially since they are at the bottom corner. I think maybe changing the color of the pop-up to a light green might make them more visible.

**Review by gahimbaref — CHANGES_REQUESTED** [review]
The toast message is more noticeable at the top. However, now when you update a phone number for the first time, the update message pops up, but if you change it again and try to update it without leaving the page first, the update messages don't show up.

File: app/templates/serviceLearning/slcProposal.html [hunk 1]
```diff
@@ -24,8 +24,8 @@ <h2 class="text-center pt-4 pb-2"> New Proposal For Service-Learning Course</h2>
             <tr data-username="" hidden>
                 <td>
                   <p class="mb-0"></p>
-                    <input type="text" style="border: none" size="12" class="form-control-sm" id="" name="courseInstructorPhone" aria-label="Instructor Phone" value="" placeholder="Phone Number" />
-                    <a class="text-decoration-none primary editButton" tabindex="0" data-id=""  id=""  type="button">Edit</a>
+                    <input type="text" style="border: none" size="14" class="form-control-sm" id="" name="courseInstructorPhone" aria-label="Instructor Phone" data-value="" value="" placeholder="Phone Number" />
```
Comment (Lyisus) [inline app/templates/serviceLearning/slcProposal.html:27]: Why is the size of the input 14? Is it due to the phone number's formatting?

File: app/static/js/base.js [hunk 2]
```diff
@@ -33,37 +32,74 @@ function msgToast(head, body){
   $("#toast-header").html(head)
   $("#toast-body").html(body)
   toastList[0].show()
+  $("#liveToast").delay(2000).fadeOut();
```
Comment (Lyisus) [inline app/static/js/base.js]: Could we make the fadeout longer? At this length is looks like something people could miss if they're a bit distracted.

File: app/templates/serviceLearning/slcProposal.html (hunk 1 above)
Comment (AndersonStettner) [reply app/templates/serviceLearning/slcProposal.html:27]: Yes, if it is 12 there are some weird cases on some devices that cut the phone number display off by one or two numbers.

**Review by gahimbaref — APPROVED** [review]

**Pull request merged**